File: windows/anti-vm.vb
'@graysuit
'https://discord.com/invite/Hu5XPGMTuk
'https://github.com/Back-X/Universal-VM-Detector
Imports System
Imports System.Management
Imports System.Windows.Forms
Public Module anti_vm
	Public Sub Main()
		Dim flag As Boolean = New ManagementObjectSearcher("SELECT * FROM Win32_PortConnector").[Get]().Count = 0
		MessageBox.Show("You are using " + If(flag, "Virtual", "Real") + " Environment", "Universal VM Detector", MessageBoxButtons.OK, If(flag, MessageBoxIcon.Exclamation, MessageBoxIcon.Asterisk))
	End Sub
End Module

File: windows/anti-vm.cs
//@graysuit
//https://graysuit.github.io
//https://github.com/Back-X/anti-vm
using System;
using System.Windows.Forms;

public class anti_vm
{
	static void Main()
	{
		bool IsVM = (new System.Management.ManagementObjectSearcher("SELECT * FROM Win32_PortConnector")).Get().Count == 0;
		MessageBox.Show
		(
		    "You are using " + (IsVM ? "Virtual" : "Real") + " Environment", 
			"Universal VM Detector", 
			MessageBoxButtons.OK, 
			(IsVM ? MessageBoxIcon.Warning : MessageBoxIcon.Information)
	    );
    }
}
